src/main/java/jose/DukeException.java

```diff
@@ -0,0 +1,15 @@
+package jose;
+
+/**
+ * A basic exception class that allows for more personalised error handling and messaging.
+ */
+public class DukeException extends Exception{
```

Comment:
This might be a mistake, perhaps you have forgotten a white space between the class Exception and open parenthesis?

I notice  the same issue in other classes too.

Reply:
Yep